LogMonitor/src/LogMonitor/EventMonitor.cpp

```diff
@@ -554,30 +560,36 @@ EventMonitor::PrintEvent(
 
             if (status == ERROR_SUCCESS)
             {
-                source = L"EventLog";
-                eventTime = Utility::FileTimeToString(fileTimeCreated);
-                eventChannel = channelName;
-                eventLevel = c_LevelToString[static_cast<UINT8>(level)];
-                eventMessage = (LPWSTR)(&m_eventMessageBuffer[0]);
-
-                std::wstring logFmt = L"<Log><Source>%s</Source><LogEntry><Time>%s</Time><Channel>%s</Channel><Level>%s</Level><EventId>%u</EventId><Message>%s</Message></LogEntry></Log>";
-                if (Utility::CompareWStrings(m_logFormat, L"JSON"))
-                {
-                    logFmt = L"{\"Source\": \"%s\",\"LogEntry\": {\"Time\": \"%s\",\"Channel\": \"%s\",\"Level\": \"%s\",\"EventId\": %u,\"Message\": \"%s\"}}";;
-                    // sanitize message
-                    std::wstring msg(m_eventMessageBuffer.begin(), m_eventMessageBuffer.end());
-                    Utility::SanitizeJson(msg);
+                pLogEntry->source = L"EventLog";
+                pLogEntry->eventTime = Utility::FileTimeToString(fileTimeCreated);
+                pLogEntry->eventChannel = channelName;
+                pLogEntry->eventLevel = c_LevelToString[static_cast<UINT8>(level)];
+                pLogEntry->eventMessage = (LPWSTR)(&m_eventMessageBuffer[0]);
+
+                std::wstring formattedEvent;
+                if (Utility::CompareWStrings(m_logFormat, L"Line")) {
+                    formattedEvent = FormatEventLineLog(m_lineLogFormat, pLogEntry);
                 }
+                else {
+                    std::wstring logFmt = L"<Log><Source>%s</Source><LogEntry><Time>%s</Time><Channel>%s</Channel><Level>%s</Level><EventId>%u</EventId><Message>%s</Message></LogEntry></Log>";
+                    if (Utility::CompareWStrings(m_logFormat, L"JSON"))

```

Comment:
They are currently on different level because JSON and XML formatting utilize the Utility::FormatString() function, which is not required in Line formatting. 

Putting all the `if`s on the same level will lead to repetition of this bit for bot XML and JSON:
```
formattedEvent = Utility::FormatString(
                        logFmt.c_str(),
                        pLogEntry->source.c_str(),
                        pLogEntry->eventTime.c_str(),
                        pLogEntry->eventChannel.c_str(),
                        pLogEntry->eventLevel.c_str(),
                        pLogEntry->eventId,
                        pLogEntry->eventMessage.c_str()
                    );
```

Reply:
Understood from your clarification on the call. I think I'm okay with avoiding the repetition.